src/main/scala/viper/gobra/frontend/PackageResolver.scala

```diff
@@ -230,22 +230,51 @@ object PackageResolver {
     } yield pkgName
   }
 
-  private lazy val pkgClauseRegex = """(?:\/\/.*|\/\*(?:.|\n)*\*\/|package(?:\s|\n)+([a-zA-Z_][a-zA-Z0-9_]*))""".r
+  private lazy val pkgClauseRegex = """(?:\/\/.*|\/\*(?:.|\n)*?\*\/|package(?:\s|\n)+([a-zA-Z_][a-zA-Z0-9_]*))""".r
 
-  private def getPackageClause(file: InputResource): Option[String] = {
-    val inputStream = file.asStream()
-    val bufferedSource = new BufferedSource(inputStream)
-    val content = bufferedSource.mkString
-    bufferedSource.close()
+  def getPackageClause(src: Source): Option[String] = {
 
     // TODO is there a way to perform the regex lazily on the file's content?
     pkgClauseRegex
-      .findAllMatchIn(content)
+      .findAllMatchIn(src.content)
       .collectFirst { case m if m.group(1) != null => m.group(1) }
   }
 
+  def getPackageId(packageName: String, src: Source, includeDirs: Vector[Path]): String = {
```

Comment:
this is hard to understand without any comments what it does / its purpose is. For example, I cannot justify for myself that the resulting string is indeed unique / an identifier. For example, there could be two different includeDirs that then have the same subdirectories. In this case, your algorithm would, I think, return non-unique identifiers.
My suggestion: I would keep the includeDir in the identifier (even though the id becomes thus longer). This obviously results in cache misses if you use non-relative includeDirs but usually they are relative to the project. I'm not sure what to do in the case of `GOPATH` as we would need a solution that works across machines

Reply:
I'm somehow tempted to just use the directory in which a package is located as identifier. If this directory is within the project then we use a relative path and in the other case we are conservative and use the absolute path

Reply:
I made it relative and gave the user a option to choose a project root directory, see `projectRoot` in `Config.scala`